linera-service/src/server.rs

```diff
@@ -558,6 +594,50 @@ async fn run(options: ServerOptions) {
                 .await
                 .unwrap();
         }
+
+        ServerCommand::EditShards {
+            server_config_path,
+            num_shards,
+            host,
+            port,
+            metrics_host,
+            metrics_port,
+        } => {
+            let mut server_config: ValidatorServerConfig =
+                util::read_json(&server_config_path).expect("Failed to read server config");
+            let mut shards = Vec::new();
+            let len = num_shards.len();
+            let num_shards = num_shards
+                .parse::<NonZeroU16>()
+                .expect("Failed to parse the number of shards");
+            let pattern = "%".repeat(len);
+
+            for i in 1u16..=num_shards.into() {
+                let index = format!("{i:0len$}", len = len);
+                let host = host.clone().replace(&pattern, &index);
+                let port = port
+                    .clone()
+                    .replace(&pattern, &index)
+                    .parse()
+                    .expect("Failed to decode port into an integers");
+                let metrics_host = metrics_host.clone().replace(&pattern, &index);
+                let metrics_port = metrics_port.clone().map(|port| {
+                    port.replace(&pattern, &index)
+                        .parse()
+                        .expect("Failed to decode metrics port into an integers")
+                });
+                let shard = ShardConfig {
+                    host,
+                    port,
+                    metrics_host,
+                    metrics_port,
+                };
+                shards.push(shard);
+            }
+            server_config.internal_network.shards = shards;
+            util::write_json(&server_config_path, &server_config)
+                .expect("Failed to write updated server config");
```

Comment:
I'm a bit worried about non-atomic saving on error conditions.  If you use `Persist`, an atomic rename will be applied automatically.

Reply:
Happy to continue the discussion separately.